kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -510,25 +534,35 @@ def kube_job_failed(self, metric, scraper_config):
 
     def kube_job_status_failed(self, metric, scraper_config):
         for sample in metric.samples:
+            job_ts = 0
             tags = [] + scraper_config['custom_tags']
             for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
                 if label_name == 'job' or label_name == 'job_name':
                     trimmed_job = self._trim_job_tag(label_value)
+                    job_ts = self._extract_job_timestamp(label_value)
                     tags.append(self._format_tag(label_name, trimmed_job, scraper_config))
                 else:
                     tags.append(self._format_tag(label_name, label_value, scraper_config))
-            self.job_failed_count[frozenset(tags)] += sample[self.SAMPLE_VALUE]
+            if job_ts != 0 and job_ts > self.failed_job_counts[frozenset(tags)].previous_run_max_ts:
+                self.failed_job_counts[frozenset(tags)].count += sample[self.SAMPLE_VALUE]
+                if job_ts > self.failed_job_counts[frozenset(tags)].current_run_max_ts:
+                    self.failed_job_counts[frozenset(tags)].current_run_max_ts = job_ts
 
```

Comment:
Could be simplified to a method like `update_current_max` on `JobCount` and then used in `kube_job_status_failed` and `kube_job_status_succeeded`. Example:

```python
            # usage in kube_job_status_failed
            self.failed_job_counts[frozenset(tags)].update(job_ts, count)

            # usage in kube_job_status_succeeded
            self.succeeded_job_counts[frozenset(tags)].update(job_ts, count)

            # impl
            class JobCount:
                def update_current_max(self, job_ts, count):
                    if job_ts != 0 and job_ts > self.previous_run_max_ts:
                        self.count += count
                        self.current_run_max_ts = max(self.current_run_max_ts, job_ts)
```

Reply:
Indeed 👍